plugins/linuxplugin/ifplugin/interface_config.go

```diff
@@ -700,82 +733,94 @@ func (plugin *LinuxInterfaceConfigurator) trackMicroservices(ctx context.Context
 			}
 		}
 
-		// First check if any microservice has terminated.
-		plugin.cfgLock.Lock()
-		for container := range plugin.microserviceByID {
-			details, err := plugin.dockerClient.InspectContainer(container)
-			if err != nil || !details.State.Running {
-				plugin.processTerminatedMicroservice(nsMgmtCtx, container)
-			}
-		}
-		plugin.cfgLock.Unlock()
+		// process changes
+		plugin.microserviceChan <- msCtx
 
-		// Now check if previously created containers have transitioned to the state "running".
-		for _, container := range created {
-			details, err := plugin.dockerClient.InspectContainer(container)
-			if err == nil {
-				if details.State.Running {
-					plugin.detectMicroservice(nsMgmtCtx, details)
-				} else if details.State.Status == "created" {
-					nextCreated = append(nextCreated, container)
-				}
-			} else {
-				plugin.Log.Debugf("Inspect container ID %v failed: %v", container, err)
-			}
+	nextRefresh:
+		// Sleep before another refresh.
+		select {
+		case <-time.After(dockerRefreshPeriod):
+			continue
+		case <-plugin.ctx.Done():
+			return
 		}
-		created = nextCreated
+	}
+}
 
-		// Finally inspect newly created containers.
-		listOpts = docker.ListContainersOptions{}
-		listOpts.All = true
-		listOpts.Filters = map[string][]string{}
-		if since != "" {
-			listOpts.Filters["since"] = []string{since}
-		}
+// HandleMicroservices handles microservice changes
+func (plugin *LinuxInterfaceConfigurator) HandleMicroservices(ctx *MicroserviceCtx) {
+	var err error
+	var newest int64
+	var containers []docker.APIContainers
+	var listOpts docker.ListContainersOptions
```

Comment:
Does the declaration need to be at the func scope? Could be just local variable.